scipy/optimize/_optimize.py

```diff
@@ -2116,7 +2116,7 @@ def terminate(warnflag, msg):
         update_l1norm = np.linalg.norm(update, ord=1)
 
     else:
-        if np.isnan(old_fval) or np.isnan(update).any():
+        if np.isnan(old_fval) or np.isnan(update_l1norm):
```

Comment:
Is it guaranteed that update_l1norm is nan whenever there‘s a nan in update?

Reply:
Yes, `np.linalg.norm` will always return `nan` if one or more elements of the vector is `nan`.